### PR TITLE
Fix some warnings from LGTM

### PR DESCRIPTION
**Pull request by Hasenn**
### **Describe your change:**
fixes minor issues : variables assigned to themselves, for - else without any break, redundant redefinitions ...

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: ciphers/enigma_machine2.py [hunk 1]
```diff
@@ -124,7 +124,8 @@ def _plugboard(pbstring: str) -> dict:
     # a) is type string
     # b) has even length (so pairs can be made)
     if not isinstance(pbstring, str):
-        raise TypeError(f"Plugboard setting isn't type string ({type(pbstring)})")
+        raise TypeError(
+            f"Plugboard setting isn't type string ({type(pbstring)})")
```
Comment (cclauss) [inline ciphers/enigma_machine2.py:128]:
```suggestion
            f"Plugboard setting isn't type string ({type(pbstring)})"
        )
```
This is what psf/black will do.

Comment (Hasenn) [reply ciphers/enigma_machine2.py:128]: fuuu i need to turn off autoformat on save

**Comment by cclauss** [comment]
Please run this PR thru psf/black.  It knows how to format Python code.

File: scheduling/shortest_job_first.py [hunk 2]
```diff
@@ -126,15 +126,20 @@ def calculate_average_times(
...
+    waiting_time = calculate_waitingtime(
+        arrival_time, burst_time, no_of_processes)
+
     bt = burst_time
     n = no_of_processes
     wt = waiting_time
     turn_around_time = calculate_turnaroundtime(bt, n, wt)
+
     calculate_average_times(waiting_time, turn_around_time, no_of_processes)
-    processes = list(range(1, no_of_processes + 1))
```
Comment (cclauss) [inline scheduling/shortest_job_first.py:135]: I do not understand this change.

Comment (Hasenn) [reply scheduling/shortest_job_first.py:135]: processes is redefined exactly the same way afterwards
without new user input

File: graphs/directed_and_undirected_(weighted)_graph.py [hunk 3]
```diff
@@ -226,7 +226,9 @@ def has_cycle(self):
                                 break
                             else:
                                 return True
-                                anticipating_nodes.add(stack[len_stack_minus_one])
+                                # TODO:The following code is unreachable.
```
Comment (cclauss) [inline graphs/directed_and_undirected_(weighted)_graph.py:229]: If code is unreachable then determine if the lines are useful.  If they are then move the return statement below them.  If they are not useful then delete them.

File: graphs/directed_and_undirected_(weighted)_graph.py [hunk 4]
```diff
@@ -226,7 +226,9 @@ def has_cycle(self):
...
+                                anticipating_nodes.add(
+                                    stack[len_stack_minus_one])
```
Comment (cclauss) [inline graphs/directed_and_undirected_(weighted)_graph.py:231]: psf/black will undo this change.

File: graphs/directed_and_undirected_(weighted)_graph.py (hunk 3 above)
Comment (Hasenn) [reply graphs/directed_and_undirected_(weighted)_graph.py:229]: i tried.. i can't read that part i don't really understand what it's supposed to do

File: graphs/directed_and_undirected_(weighted)_graph.py (hunk 4 above)
Comment (Hasenn) [reply graphs/directed_and_undirected_(weighted)_graph.py]: yeah autopep8 did that, i've just pushed after running black

**Review by cclauss — APPROVED** [review]
LGTM!
